File: snippets/fsharp/System/Type/DefaultBinder/type_defaultbinder.fs
module type_defaultbinder

// <Snippet1>
open System
open System.Reflection

type MyClass() =
    member _.HelloWorld() =
        printfn "Hello World"

try
    let defaultBinder = Type.DefaultBinder
    let myClass = MyClass()
    // Invoke the HelloWorld method of MyClass.
    myClass.GetType().InvokeMember("HelloWorld", BindingFlags.InvokeMethod, defaultBinder, myClass, [||])
    |> ignore
with e ->
    printfn $"Exception: {e.Message}"
// </Snippet1>

File: snippets/csharp/System/Type/DefaultBinder/type_defaultbinder.cs
// <Snippet1>
using System;
using System.Reflection;

public class MyDefaultBinderSample
{
    public static void Main()
    {
        try
        {
            Binder defaultBinder = Type.DefaultBinder;
            MyClass myClass = new();
            // Invoke the HelloWorld method of MyClass.
            myClass.GetType().InvokeMember("HelloWorld", BindingFlags.InvokeMethod,
                defaultBinder, myClass, []);
        }
        catch (Exception e)
        {
            Console.WriteLine("Exception :" + e.Message);
        }
    }

    class MyClass
    {
        public void HelloWorld() => Console.WriteLine("Hello World");
    }
}
// </Snippet1>
